solutions/CSES Planets Cycles.mdx

```diff
@@ -0,0 +1,92 @@
+---
+id: cses-1751
+source: CSES 
+title: Planets Cycles
+author: Maggie Liu
+---
+
+Define the number of teleportations starting from a planet as the $\texttt{cyclelength}$ of that planet. For each planet that hasn't been visited, we want to find its $\texttt{cyclelength}$. Call the planet we are performing $\texttt{dfs}$ from the $\textit{start}$. As we perform $\texttt{dfs}$ from the $\textit{start}$, keep track of the planets seen, in order, in the $\texttt{cycle}$ queue. When we reach a planet that has already been visited (call this planet the $\textit{repeat}$), we add the $\texttt{cyclelength}$ of the $\textit{repeat}$ to the current $\texttt{step}$ count. 
+
+Once we have $\texttt{cycle}$ and $\texttt{steps}$ (the $\texttt{cyclelength}$ of the $\textit{start}$) we can calculate the $\texttt{cyclelength}$ of all the planets in this $\texttt{cycle}$. We know the $\textit{repeat}$ will always be the planet at the end of the $\texttt{cycle}$, but there are two cases for the $\textit{repeat}$. 
```

Comment:
```suggestion
Once we have $\texttt{cycle}$ and $\texttt{steps}$ (the $\texttt{cyclelength}$ of the $\textit{start}$) we can calculate the $\texttt{cyclelength}$ of all the planets in this $\texttt{cycle}$. We know the $\textit{repeat}$ will always be the planet at the end of the $\texttt{cycle}$, but it may appear elsewhere as well. We can break this down into two cases:
```